streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/SuppressedImpl.java

```diff
@@ -37,12 +36,12 @@ public SuppressedImpl(final Duration suppressionTime,
                           final boolean suppressTombstones) {
         this.timeToWaitForMoreEvents = suppressionTime == null ? DEFAULT_SUPPRESSION_TIME : suppressionTime;
         this.timeDefinition = timeDefinition == null ? (context, anyKey) -> context.timestamp() : timeDefinition;
-        this.bufferConfig = bufferConfig == null ? DEFAULT_BUFFER_CONFIG : bufferConfig;
+        this.bufferConfig = bufferConfig == null ? DEFAULT_BUFFER_CONFIG : (BufferConfigImpl) bufferConfig;
         this.suppressTombstones = suppressTombstones;
     }
 
-    interface TimeDefinition<K> {
-        long time(final ProcessorContext context, final K key);
+    BufferConfigImpl getBufferConfig() {
+        return bufferConfig;
```

Comment:
Should we cast here and keep `BufferConfig bufferConfig` as member type?

Reply:
This made me realize that I named them `Impl` when I meant to name them `Internal`. In other words, both `SuppressedInternal` and `BufferConfigInternal` to indicate that these are the internal interfaces.